src/admin/components/forms/field-types/Number/index.tsx

```diff
@@ -145,7 +145,7 @@ const NumberField: React.FC<Props> = (props) => {
           filterOption={(option, rawInput) => {
             // eslint-disable-next-line no-restricted-globals
             const notANumber = rawInput.trim() === '' || isNaN(Number(rawInput));
```

Comment:
Maybe we should use this [isNumber](https://github.com/payloadcms/payload/blob/master/src/utilities/isNumber.ts) that we are using elsewhere?

Reply:
@JarrodMFlesch Done! I've adjusted the isNumber function as well, as it didn't handle the edge-case where JavaScript is dumb and converts "   " into 0